src/context.cpp

```diff
@@ -221,8 +221,28 @@ namespace librealsense
 #endif //BUILD_WITH_DDS
     }
 
-    std::vector<std::shared_ptr<device_info>> context::query_devices(int mask) const
+    static unsigned calc_mask( nlohmann::json const & settings, unsigned requested_mask, unsigned * p_device_mask = nullptr )
     {
+        // The 'device-mask' in our settings can be used to disable certain device types from showing up in the context:
+        unsigned device_mask = RS2_PRODUCT_LINE_ANY;
+        unsigned mask = requested_mask;
+        if( rsutils::json::get_ex( settings, "device-mask", &device_mask ) )
+        {
+            // The normal bits enable, so enable only those that are on
+            mask &= device_mask & ~RS2_PRODUCT_LINE_SW_ONLY;
+            // But the above turned off the SW-only bits, so turn them back on again
+            if( (device_mask & RS2_PRODUCT_LINE_SW_ONLY) || (requested_mask & RS2_PRODUCT_LINE_SW_ONLY) )
+                mask |= RS2_PRODUCT_LINE_SW_ONLY;
+        }
+        if( p_device_mask )
+            *p_device_mask = device_mask;
```

Comment:
Why do we need to output `device_mask`? If only for logging then we should log from this function.

Reply:
I didn't want to log from this function, it's called many many times inside from the device-update mechanism...

Reply:
I missed the call in `create_devices` thought it was called only from `query_devices`